src/leap/mail/imap/mailbox.py

```diff
@@ -322,7 +324,34 @@ def addMessage(self, message, flags, date=None):
         :return: a deferred that will be triggered with the UID of the added
                  message.
         """
-        return self.collection.add_raw_message(message, flags, date)
+        # TODO should raise ReadOnlyMailbox if not rw.
+        # TODO have a look at the cases for internal date in the rfc
+        # XXX we could treat the message as an IMessage from here
+
+        if isinstance(message, (cStringIO.OutputType, StringIO.StringIO)):
```

Comment:
Can it be missing the `import StringIO`? I only see the cStringIO import. Or could you use cStringIO instead of StringIO?